Source/Core/loadImageFromTypedArray.js

```diff
@@ -31,22 +31,30 @@ define([
             type : format
         });
 
-        var blobUrl = window.URL.createObjectURL(blob);
-        var resource = new Resource({
-            url: blobUrl,
-            request: request
-        });
-        return resource.fetchImage({
-            flipY : flipY,
-            preferImageBitmap : true
-        })
-            .then(function(image) {
-                window.URL.revokeObjectURL(blobUrl);
-                return image;
-            })
-            .otherwise(function(error) {
-                window.URL.revokeObjectURL(blobUrl);
-                return when.reject(error);
+        return Resource.supportsImageBitmapOptions()
+            .then(function(result) {
+                if (result) {
+                    return when(Resource.createImageBitmapFromBlob(blob, flipY));
+                }
+
+                var blobUrl = window.URL.createObjectURL(blob);
+                var resource = new Resource({
+                    url: blobUrl,
+                    request: request
+                });
+
+                return resource.fetchImage({
+                    flipY : flipY
+                })
+                    .then(function(image) {
+                        window.URL.revokeObjectURL(blobUrl);
+                        return image;
+                    }, function(error) {
+                    })
+                    .otherwise(function(error) {
+                        window.URL.revokeObjectURL(blobUrl);
+                        return when.reject(error);
+                    });
```

Comment:
Why does this `then` have an empty `function(error){}` that doesn't catch or do anything with the error? You shouldn't need it if you have an `otherwise` that actually catches the error.

Also, you can avoid nesting these functions by having just one top level sequence of `then`'s. In the case where it returns early, the subsequent thens can check if the returned thing is undefined, and then exit early as well. `Resource.prototype.fetchImage` has an example of this pattern.